### PR TITLE
Fixes to Paladin spell procs

### PR DESCRIPTION
**Pull request by GuybrushGit**
- https://github.com/elysium-project/server/issues/736 JoC and JoR now proc as melee spells instead of magic spells.
- https://github.com/elysium-project/server/issues/1963 Redoubt now procs on what they should according to spell data.
- https://github.com/elysium-project/server/issues/1601 Damage shields like Retribution are now affected by modifiers.
- https://github.com/elysium-project/server/issues/604 https://github.com/elysium-project/server/issues/2165 https://github.com/elysium-project/server/issues/2458 Self damage like Goblin Sapper will now trigger procs. Note: Spells with no damage class like the Runes will continue not proccing anything, this is intended.

### CODE REVIEW

**Comment by Mostprovidedlinks** [comment]
Ty guybushgit

**Comment by connormcgregor** [comment]
But it shows dark runes have a shadow damage school and can be resisted.What do you mean runes dont have a damage class.The second effect is dark rune is thesame as goblin sapper charge which damage you.
First effect of rune is energize which has nothing to do with damage. First effect of sapper is damage your opponent.

**Comment by connormcgregor** [comment]
Why would it not be a damage class? I see shadow school

**Comment by connormcgregor** [comment]
School Shadow
Mechanic n/a Dispel type n/a Cost None Range 0 yards 
(Self Only) Cast time Instant Cooldown n/a
Effect #1(30) Energize 
Value: 900
Effect #2(2) School Damage (Shadow) 
Value: 600

**Comment by Itembugs** [comment]
Can uou add this
https://github.com/elysium-project/server/issues/1492

**Comment by GuybrushGit** [comment]
@Itembugs no

**Comment by Itembugs** [comment]
When will vengeance be fixed with storm gauntlets?

**Comment by Mostprovidedlinks** [comment]
Please add this

https://github.com/elysium-project/database/issues/75

So consecration affects wrath of cenarius

File: src/game/Spells/Spell.cpp [hunk 1]
```diff
@@ -1231,6 +1238,12 @@ void Spell::DoAllEffectOnTarget(TargetInfo *target)
         // item procs should probably trigger for all classes, e.g. Judgement of Wisdom, but many issues to test for first. Paladin exclusive for now.
         if (m_CastItem && m_caster->getClass() == CLASS_PALADIN && (procEx & PROC_EX_CRITICAL_HIT) && !(m_spellInfo->AttributesEx3 & SPELL_ATTR_EX3_CANT_TRIGGER_PROC))
             m_canTrigger = true;
+        // JoR and JoC: Paladin melee spells trigger melee procs instead of magic
+        if ((m_spellInfo->IsFitToFamilyMask<CF_PALADIN_JUDGEMENT_OF_RIGHTEOUSNESS>() && m_spellInfo->SpellIconID == 25) || m_spellInfo->SpellIconID == 561)
```
Comment (thepparker) [inline src/game/Spells/Spell.cpp]: JoC the only spell with icon ID 561?

Comment (GuybrushGit) [reply src/game/Spells/Spell.cpp]: Bad copy paste. Fixed it.

**Pull request merged**